schema/appmodel/trigger.schema.xml

```diff
@@ -247,11 +247,12 @@
  <class name="TAMakerADCSimpleWindowAlgorithm">
   <superclass name="TAAlgorithm"/>
   <attribute name="adc_threshold" description="Threshold for the ADC integral algorithm" type="u32" init-value="10000" is-not-null="yes"/>
-  <attribute name="window_length" description="Window length (in ticks) for the ADC integral algorithm" type="u32" init-value="10000" is-not-null="yes"/>
+  <attribute name="window_length" description="Window length (in ticks) for the algorithm" type="u32" init-value="10000" is-not-null="yes"/>
```

Comment:
Is window_length a property of all TAAlgorithms? Should this attribute be moved to the base class?

Reply:
Hi @eflumerf, it is not. Most of our algorithms do use a window_length, but not all of them (e.g. bundle algorithm, prescale algorithm, DBSCAN too I think), which is why I didn't add it to the base class.

Reply:
@ArturSztuc Could there be another class level, e.g., `TAWindowAlgorithm` that is derived from `TAAlgorithm` and has `window_length`?

Reply:
@aeoranday I actually don't know, but I imagine this should be possible